core/shared/src/main/scala/zio/Fiber.scala

```diff
@@ -403,13 +403,15 @@ object Fiber extends FiberPlatformSpecific {
     /**
      * Generates a fiber dump.
      */
-    final def dump: UIO[Fiber.Dump] =
+    final def dump(withTrace: Boolean): UIO[Fiber.Dump] =
       for {
-        name   <- self.getRef(Fiber.fiberName)
-        id     <- self.id
-        status <- self.status
-        trace  <- self.trace
-      } yield Fiber.Dump(id, name, status, trace)
+        name       <- self.getRef(Fiber.fiberName)
+        id         <- self.id
+        status     <- self.status
+        trace      <- if (withTrace) self.trace.map(Some(_)) else UIO(None)
+        ch         <- self.children
+        childDumps <- ZIO.foreach(ch)(c => c.asInstanceOf[Fiber.Runtime[_, _]].dump(withTrace))
```

Comment:
All the children should be runtime fibers. You can change the type of `children` in `Fiber.Descriptor` to `Fiber.Runtime[Any, Any]` and the return type of `children` in `FiberContext` to `Fiber.Runtime[Any, Any]` to remove the need for this cast.